src/main/java/com/powsybl/openloadflow/ac/nr/NewtonRaphson.java

```diff
@@ -63,7 +65,33 @@ public static List<Pair<Equation<AcVariableType, AcEquationType>, Double>> findL
                 .collect(Collectors.toList());
     }
 
-    private NewtonRaphsonStatus runIteration(StateVectorScaling svScaling, MutableInt iterations) {
+    public static void reportLargestMismatch(Reporter reporter, EquationSystem<AcVariableType, AcEquationType> equationSystem, double[] mismatch, LfNetwork network, int iteration) {
+        Map<AcEquationType, Optional<Pair<Equation<AcVariableType, AcEquationType>, Double>>> mismatchEquations = equationSystem.getIndex().getSortedEquationsToSolve().stream()
+                .map(equation -> Pair.of(equation, mismatch[equation.getColumn()]))
+                .collect(Collectors.groupingBy(e -> e.getKey().getType(), Collectors.maxBy(Comparator.comparingDouble(e -> Math.abs(e.getValue())))));
+
+        List<AcEquationType> acEquationTypes = List.of(AcEquationType.BUS_TARGET_P, AcEquationType.BUS_TARGET_Q, AcEquationType.BUS_TARGET_V);
+
+        for (AcEquationType acEquationType : acEquationTypes) {
+            mismatchEquations.get(acEquationType)
+                    .ifPresent(equationPair -> {
+                        Equation<AcVariableType, AcEquationType> equation = equationPair.getKey();
+                        double equationMismatch = equationPair.getValue();
+                        int elementNum = equation.getElementNum();
+                        String elementId = equation.getElement(network).map(LfElement::getId).orElse("?");
+                        int busVRow = equationSystem.getVariable(elementNum, AcVariableType.BUS_V).getRow();
+                        int busPhiRow = equationSystem.getVariable(elementNum, AcVariableType.BUS_PHI).getRow();
+                        double busV = equationSystem.getStateVector().get(busVRow);
+                        double busPhi = equationSystem.getStateVector().get(busPhiRow);
+                        if (LOGGER.isTraceEnabled()) {
```

Comment:
So it does not only report? It also log using SLF4J. Is it intended to replace the previous one (X largest mismatch) or be in addition to?

Reply:
The main purpose was for it to be in addition to the existing one. But now that you mentioned it, maybe it could replace the previous one as it logs a more digestible information...